data/org.eclipse.birt.report.data.oda.jdbc.ui/src/org/eclipse/birt/report/data/oda/jdbc/ui/editors/SQLDataSetEditorPage.java

```diff
@@ -1075,4 +1123,149 @@ private void insertTreeItemText() {
+		this.findQueryTextWholeWord.setSelection(false);
+		this.findQueryTextWholeWord.setLayoutData(wwLayoutData);
+		this.findQueryTextWholeWord.setEnabled(true);
+	}
+
+	private void setupFindQueryTextButtons(Group group) {
+
+		GridData fwButtonFindTextLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		Button fwButtonFindText = new Button(group, SWT.BUTTON1);
+		fwButtonFindText.setText(
+				FIND_DIRECTION_SYMBOLE_FORWARD + " " + JdbcPlugin.getResourceString("tablepage.querytext.find.button.forward")); //$NON-NLS-1$
+		fwButtonFindText.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.button.forward.tooltip")); //$NON-NLS-1$
+		fwButtonFindText.setLayoutData(fwButtonFindTextLayoutData);
+		fwButtonFindText.setEnabled(true);
+		// Add listener to the find query button "forward"
+		fwButtonFindText.addSelectionListener(new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				findQueryTextForward();
+			}
+		});
+
+		GridData bwButtonFindTextLayoutData = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+		Button bwButtonFindText = new Button(group, SWT.BUTTON1);
+
+		bwButtonFindText.setText(FIND_DIRECTION_SYMBOLE_BACKWARD + " " //$NON-NLS-1$
+				+ JdbcPlugin.getResourceString("tablepage.querytext.find.button.backward"));
+		bwButtonFindText.setToolTipText(JdbcPlugin.getResourceString("tablepage.querytext.find.button.backward.tooltip")); //$NON-NLS-1$
+		bwButtonFindText.setLayoutData(bwButtonFindTextLayoutData);
+		bwButtonFindText.setEnabled(true);
+		// Add listener to the find query button "forward"
+		bwButtonFindText.addSelectionListener(new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				findQueryTextBackward();
+			}
+		});
+	}
+
+	private void setupLabelFindQueryTextGroup(int findMode) {
+		String groupLabel = JdbcPlugin.getResourceString("tablepage.querytext.find.label");
+
+		// handle the default color of the group label
+		if (this.findGroupLabelOriginalColor == null)
+			this.findGroupLabelOriginalColor = new Color(findQueryTextGroup.getForeground().getDevice(),
+					findQueryTextGroup.getForeground().getRGBA());
+		Color fontColor = null;
+
+		// forward search end
+		if (findMode == 1) {
+			fontColor = new Color(Display.getCurrent(), 255, 0, 0);
+			groupLabel += ", " + JdbcPlugin.getResourceString("tablepage.querytext.find.forward.unlocated.label");
+			// backward search end
+		} else if (findMode == 2) {
+			fontColor = new Color(Display.getCurrent(), 255, 0, 0);
+			groupLabel += ", " + JdbcPlugin.getResourceString("tablepage.querytext.find.backward.unlocated.label");
+		} else {
+			fontColor = new Color(this.findGroupLabelOriginalColor.getDevice(),
+					this.findGroupLabelOriginalColor.getRGBA());
+		}
+
+		findQueryTextGroup.setForeground(fontColor);
+		findQueryTextGroup.setText(groupLabel);
+		fontColor.dispose();
```

Comment:
Wow, does this even work?   Setting the color doesn't copy so you can't dispose something that's in use.   I have a feeling that Color doesn't actually do anything when disposed, on some or maybe all platforms, but in any case this is a super bad pattern to follow.

Reply:
What is your suggestion for the Color-dispose?

Reply:
The dispose will work!
Because that this is the reason why I copy the color-object to a new object value.
If I use the same reference without copy then the dispose destroy/dispose really the color object and it isn't available any more.

Reply:
If disposed did something then the widget using the color would be broken.

Please look at how the platform is doing the same thing as I've referenced elsewhere.

Reply:
I will change it.

Reply:
Changed